src/process/ProcessManagerImpl.cpp

```diff
@@ -775,12 +861,14 @@ void
 ProcessManagerImpl::maybeRunPendingProcesses()
 {
     ZoneScoped;
+    checkInvariants();
     if (mIsShutdown)
     {
         return;
     }
     std::lock_guard<std::recursive_mutex> guard(mProcessesMutex);
-    while (!mPending.empty() && gNumProcessesActive < mMaxProcesses)
+    while (!mPending.empty() &&
+           getNumRunningOrShuttingDownProcesses() < mMaxProcesses)
```

Comment:
nit: `if (fs::exists(i->mImpl->mOutFile))` below should probably be `if (!i->mImpl->mOutFile.empty() && fs::exists(i->mImpl->mOutFile))` : that variant of `fs::exists` can throw and per standard I can't tell what the behavior is on empty filename as it's not a valid path.

Reply:
Good catch, fixed.